libraries/AP_HAL_Linux/Util.h

```diff
@@ -48,7 +48,7 @@ class Util : public AP_HAL::Util {
      */
     void set_system_clock(uint64_t time_utc_usec);
     const char* get_custom_log_directory() { return custom_log_directory; }
-    const char* get_custom_terrain_directory() { return custom_terrain_directory; }
+    const char* get_custom_terrain_directory() const final { return custom_terrain_directory; }
```

Comment:
> Class A has method foo, Class B has method fooo, it isn't overriding anything.

Exactly. I wanted to demonstrate the it detects the same kind of bugs that override is supposed to detect.

> An overriding method is implicitly virtual

From the parent class, yes. "final" makes the method on this class not be "overridable". It's actually used as tip to the compiler that when it sees that method and the object in which it's operating is of that type it knows that it is the last possible override. This helps recent versions of compilers to devirtualize some methods.

Since "final" only makes sense if the method on the parent class is virtual, it was used on the example above to detect a bug in which we were not really overriding the parent method due to a typo.